File: DesignModel/warn_unused_result/ViewController.m
//
//  ViewController.m
//  warn_unused_result
//
//  Created by 侯森魁 on 2017/6/5.
//  Copyright © 2017年 小虎金融. All rights reserved.
//

#import "ViewController.h"

@interface ViewController ()

@end

@implementation ViewController

/*
 warn_unused_result 这个关键字的含义：如果某个函数使用了这个关键字，那么函数在被调用的时候，要检查或者使用返回值，否则编译器会进行警告。
 使用场合：在把一些功能封装起来(或者SDK的编写)时候，如果对返回值的使用比较重要，那么使用这个关键字提醒编译器要检查返回值是否被利用。
 
 
 */
- (BOOL)TestFunc:(NSInteger)num __attribute__ ((warn_unused_result)){
    
    return num > 0?YES:NO;
}

- (void)viewDidLoad {
    [super viewDidLoad];
    BOOL result  =  [self TestFunc:10];
    
    
    // Do any additional setup after loading the view, typically from a nib.
}


- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}


@end

File: DesignModel/FunctionTest/ViewController.swift
//
//  ViewController.swift
//  FunctionTest
//
//  Created by 史蒂夫 on 2017/6/1.
//  Copyright © 2017年 小虎金融. All rights reserved.
//

import UIKit

class ViewController: UIViewController {

    lazy var materialTableview :UITableView =  {
        
        weak var weakSelf  = self
        var tableView  =  UITableView()
        
        tableView.frame = weakSelf!.view.bounds
        
        tableView.backgroundColor = UIColor.redColor()
        return tableView
        
    }()
    lazy var tableView = UITableView()
        
    override func viewDidLoad() {
        super.viewDidLoad()
        greet("housenkui", day: "Sunday")
        
        greet01("housenkui", "Sunday")
        
        view.addSubview(self.materialTableview)

        // Do any additional setup after loading the view, typically from a nib.
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }

    func greet( person:String,day:String){
        
        print("Hello \(person),today is \(day)")
    }
    
    func greet01(_ person:String,_ day:String){
        
        print("Hello \(person),today is \(day)")
    }
    
    

}

